dal/src/main/resources/ee/tlu/evkk/dal/dao/TextDao.xml

```diff
@@ -9,66 +9,105 @@
     </select>
 
-    <select id="findTextIDsByCorpusID" resultType="java.lang.String">
-        select text_id
-        from core.text_property
-        where core.text_property.property_value = #{corpusId} and core.text_property.property_name = 'korpus';
-    </select>
+  <select id="findTextIdsByCorpusId" resultType="java.lang.String">
+    select text_id
+    from core.text_property
+    where core.text_property.property_value = #{corpusId}
+      and core.text_property.property_name = 'korpus';
+  </select>
 
-    <select id="findTextIDandTitleByCorpusID" resultType="java.lang.String">
-        select array_to_json(array_agg(row_to_json(t)))
-        from (
-            select tp1.text_id,
-            (select tp2.property_value FROM core.text_property as tp2 where tp2.property_name='title' and tp2.text_id = tp1.text_id)
-        from core.text_property as tp1
-        where tp1.property_value = #{corpusId} and tp1.property_name = 'korpus'
-        ) t
-    </select>
+  <select id="findTextIdAndTitleByCorpusId" resultType="java.lang.String">
+    select array_to_json(array_agg(row_to_json(t)))
+    from (select tp1.text_id,
+                 (select tp2.property_value
+                  FROM core.text_property as tp2
+                  where tp2.property_name = 'title'
+                    and tp2.text_id = tp1.text_id)
+          from core.text_property as tp1
+          where tp1.property_value = #{corpusId}
+            and tp1.property_name = 'korpus') t
+  </select>
 
-    <select id="textTitleQueryByParameters" resultType="java.lang.String">
-        select array_to_json(array_agg(row_to_json(t)))
-        from (
-            select p2.text_id, p2.property_value
-            from core.text_property as p2
-            <if test="loendurHelperid.length != 0">
-                join core.text as p3 on p2.text_id = p3.id
-                join core.text_processor_result as p4 on p3.hash = p4.text_hash
-            </if>
-            <foreach item="helperid" collection="helperid" separator=" ">join core.text_property as ${helperid.tabel} on p2.text_id = ${helperid.tabel}.text_id</foreach>
-            where p2.property_name = 'title'
-            <foreach item="helperid" collection="helperid" separator=" "> and ${helperid.tabel}.property_name = #{helperid.parameeter} and ${helperid.tabel}.property_value in <foreach item="v" collection="helperid.vaartus" separator=", " open="(" close=")">#{v}</foreach></foreach>
-            <if test="loendurHelperid.length != 0">
-             and p4.type = 'CHARS_WORDS_SENTENCES'
-                <foreach item="loendurHelperid" collection="loendurHelperid" separator=" ">
-                     and cast(p4.result::json->#{loendurHelperid.parameeter} as TEXT) = #{loendurHelperid.vaartus}
-                </foreach>
-            </if>
-        ) t
-    </select>
+  <select id="detailedTextQueryByParameters" resultType="java.lang.String">
```

Comment:
Päring on omajagu keerukas - peaks kontrollima et see käib vastu indekseid ja et `$` pandud muutujad on kindlasti sellised, mida kasutaja sisendiga mõjutada ei saa. Ehk siis midagi otseselt valesti ei ole, lihtsalt oleks vaja paaris asjas veenduda.

Reply:
text_processor_result tabelist type ja result peaks olema ainsad, mida kasutatakse where klauslites ja selectides ning mis pole indekseeritud. ```$``` on pandud ainult .table propertyle, mille väärtused on igale parameetrile service kihis hardcoded.

Reply:
Kas oleks äkki otstarbekas indeksid ära teha et otsingud kiiremini käiks?

Reply:
result columnit ei saa indexida sest seal liiga palju andmeid (index row requires bytes maximum size is 8191), aga type columnile lisasin